File: lab02_stack/example03/scripts/publisher_node.py
#!/usr/bin/env python

import rospy
import math
from geometry_msgs.msg import PointStamped

def talker():

    pub = rospy.Publisher('chatter', PointStamped, queue_size=10)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10) # 10hz

    radius = 5
    period = 5

    msg = PointStamped()

    starting_time = rospy.Time.now().to_sec()

    while not rospy.is_shutdown():

        msg.header.stamp = rospy.Time.now()

        msg.point.x = radius * math.cos(2 * math.pi * (msg.header.stamp.to_sec() - starting_time)/period)
        msg.point.y = radius * math.sin(2 * math.pi * (msg.header.stamp.to_sec() - starting_time)/period)
        msg.point.z = 0.0

        pub.publish(msg)
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: lab02_stack/example02/src/publisher_node.cpp
#include "ros/ros.h"
#include "geometry_msgs/PointStamped.h"

int main(int argc, char **argv)
{

    ros::init(argc, argv, "talker");

    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<geometry_msgs::PointStamped>("/publish_point", 1000);

    ros::Rate loop_rate(10);

    int radius = 5;
    int period = 5;
    geometry_msgs::PointStamped msg;

    double starting_time =ros::Time::now().toSec();

    while (ros::ok())
    {

        msg.header.stamp = ros::Time::now();

        //Publish the position on a 5 unit radius-circle
        msg.point.x = radius * cos(2 * M_PI * (msg.header.stamp.toSec() - starting_time)/period);
        msg.point.y = radius * sin(2 * M_PI * (msg.header.stamp.toSec() - starting_time)/period);
        msg.point.z = 0;

        pub.publish(msg);

        ros::spinOnce();

        loop_rate.sleep();
    }


    return 0;
}

